src/panels/lovelace/editor/card-editor/hui-card-editor.ts

```diff
@@ -0,0 +1,264 @@
+import {
+  html,
+  css,
+  LitElement,
+  TemplateResult,
+  CSSResult,
+  customElement,
+  property,
+} from "lit-element";
+
+import yaml from "js-yaml";
+
+import "@material/mwc-button";
+import { HomeAssistant } from "../../../../types";
+import { LovelaceCardConfig } from "../../../../data/lovelace";
+import { LovelaceCardEditor } from "../../types";
+import { getCardElementTag } from "../../common/get-card-element-tag";
+
+import "../../components/hui-yaml-editor";
+// This is not a duplicate import, one is for types, one is for element.
+// tslint:disable-next-line
+import { HuiYamlEditor } from "../../components/hui-yaml-editor";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { EntityConfig } from "../../entity-rows/types";
+
+declare global {
+  interface HASSDomEvents {
+    "entities-changed": {
+      entities: EntityConfig[];
+    };
+    "config-changed": {
+      config: LovelaceCardConfig;
+    };
+  }
+}
+
+@customElement("hui-card-editor")
+export class HuiCardEditor extends LitElement {
+  @property() public hass?: HomeAssistant;
+  @property() public value?;
+
+  @property() private _yaml?: string | undefined;
+  @property() private _configElement?: LovelaceCardEditor;
+  @property() private _configElType?: string;
+  @property() private _GUImode: boolean = true;
+  // Error: Configuration broken - do not save
+  @property() private _error?: string;
+  // Warning: GUI editor can't handle configuration - ok to save
+  @property() private _warning?: string;
+  @property() private _loading: boolean = false;
+
+  public get yaml(): string {
+    return this._yaml || "";
+  }
+  public set yaml(_yaml: string) {
+    this._yaml = _yaml;
+    this._updateConfigElement();
+    // TODO: Defer this to next redraw? How?
+    setTimeout(() => this._yamlEditor.codemirror.refresh(), 1);
+    fireEvent(this, "config-changed", { config: this.config! });
```

Comment:
If we want to follow how DOM elements fire change events, we should not trigger when the value is set from the outside, but only when it is changed from within the element.

Reply:
Otherwise we can get into loops.

Reply:
I finally realized the obvious way to fix this.
Yaml is still the main source of truth, but each time it is changed, it's also parsed into a configuration. If invalid, the old configuration is still stored, but an error is sent to the host element.
When updating the configuration (which is now known as `value`) from outside, it's checked against this stored value, and if it's equal, nothing is changed. Otherwise, it replaces the yaml.

Returning and being fed back bad values avoided. Loops gone. I hope.